packages/babel-helper-member-expression-to-functions/src/index.js

```diff
@@ -100,6 +100,34 @@ const handle = {
       return;
     }
 
+    // { KEY: MEMBER } = OBJ -> { KEY: _destructureSet(MEMBER) } = OBJ
+    // { KEY: MEMBER = _VALUE } = OBJ -> { KEY: _destructureSet(MEMBER) = _VALUE } = OBJ
+    // {...MEMBER} -> {..._destructureSet(MEMBER)}
+    //
+    // [MEMBER] = ARR -> [_destructureSet(MEMBER)] = ARR
+    // [MEMBER = _VALUE] = ARR -> [_destructureSet(MEMBER) = _VALUE] = ARR
+    // [...MEMBER] -> [..._destructureSet(MEMBER)]
+    if (
+      // { KEY: MEMBER } = OBJ
+      (parentPath.isObjectProperty({ value: node }) &&
+        parentPath.parentPath.isObjectPattern()) ||
+      // { KEY: MEMBER = _VALUE } = OBJ
+      (parentPath.isAssignmentPattern({ left: node }) &&
+        parentPath.parentPath.isObjectProperty({ value: parent }) &&
+        parentPath.parentPath.parentPath.isObjectPattern()) ||
```

Comment:
Nit: I think we can actually join a few of the conditions here. `parentPath.isAssignmentPattern()` is sufficient to check for both array pattern and object pattern, we don't need to consult the grand parent.

Reply:
We can move these checks to be assertions: (`t.assertObjectProperty(parentPath.parentPath, { value: parent })`

Reply:
@nicolo-ribaudo what is the rationale behind using `assertions` instead of `is`?

Reply:
Just that with an assertion is clear that if this is an assignment expression, the parent will always be an object/array destructing

Reply:
Oh ok nevermind, we can't assert two different parent types.